test/tools/yulopti.cpp

```diff
@@ -87,37 +89,72 @@ class YulOpti
 		}.printErrorInformation(_errors);
 	}
 
-	void parse(string const& _input)
+	shared_ptr<Object> getSubObject(shared_ptr<Object> const& _object, string const& _qualifiedPath)
+	{
+		if (_qualifiedPath.empty() || _qualifiedPath == _object->name.str())
+			return _object;
+
+		yulAssert(
+			boost::algorithm::starts_with(_qualifiedPath, _object->name.str() + "."),
+			"Assembly object not found."
+		);
+
+		const auto subObjectPath = _qualifiedPath.substr(_object->name.str().length() + 1);
+		const auto subObjectName = subObjectPath.substr(0, subObjectPath.find_first_of('.'));
```

Comment:
```suggestion
		const string subObjectPath = _qualifiedPath.substr(_object->name.str().length() + 1);
		const string subObjectName = subObjectPath.substr(0, subObjectPath.find_first_of('.'));
```
I don't see any benefit of using `auto` here.

Reply:
Also, `const` qualifier has to come after the type as per our coding style, so in this case, `string const` instead of `const string`.